client/src/pages/Messenger.jsx

```diff
@@ -1,75 +1,103 @@
 import { makeStyles } from "@material-ui/core/styles";
-import CssBaseLine from "@material-ui/core/CssBaseline";
-import Hidden from "@material-ui/core/Hidden";
-import * as Components from '../components/Components';
+import * as Components from "../components/Components";
+import { useHistory } from "react-router-dom";
+import cookie from "react-cookies";
+import clsx from "clsx";
 
-const useStyles = makeStyles(theme => ({
-	root: {
-		minHeight: "100vh",
-		// "& .MuiInput-underline:before": {
-		// 	borderBottom: "1.2px solid rgba(0, 0, 0, 0.2)"
-		// }
+const drawerWidth = "35vw";
 
-	},
-	contactsListContainer: {
-		width: "100%",
-		alignSelf: "center",
-		display: "flex",
-		overflow: "auto",
-		flexGrow: 1,
-		flexDirection: "column",
-		padding: "16px",
-		paddingTop: 0,
-	},
-	dividedPages: {
-		height: "100vh",
-		display: "flex",
-		flexDirection: "column",
-	},
-	chats: {
-		height: "100vh",
-	},
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    width: "100vw",
+    height: "100vh",
+  },
+  drawer: {
+    width: drawerWidth,
+    height: "100%",
+  },
+  chatChannel: {
+    width: "100vw",
+    height: "100%",
+    flexGrow: 1,
+    display: "flex",
+    flexDirection: "column",
+    paddingLeft: 15,
+    marginLeft: `-${drawerWidth}`,
+  },
+  chatChannelShift: {
+    width: `calc(100% - ${drawerWidth})`,
+    transition: theme.transitions.create("margin", {
+      easing: theme.transitions.easing.easeOut,
+      duration: theme.transitions.duration.enteringScreen,
+    }),
+    marginLeft: 0,
+  },
 }));
 
+const useAuthenticate = () => {
+  const history = useHistory();
 
-export default function Messenge(props) {
-	const classes = useStyles(props);
-	return (
-		<Grid container component="main" className={classes.root} width="100vw">
-			<CssBaseLine />
-			<Grid item xs={false} sm={4} md={5} spacing={1}>
-				<Hidden xsDown>
-					<Box className={classes.dividedPages}>
-						<Components.UserHeaderComponent />
-						<Components.ChatAndSearchbarComponent />
-						{/* Contacts List */}
-						<Box className={classes.contactsListContainer}>
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-							<Components.ContactsListComponent />
-						</Box>
-					</Box>
-				</Hidden>
-			</Grid>
+  const authenticate = async () => {
+    const res = await fetch("/auth/login", {
+      method: "POST",
+      headers: {
+        Authentication: cookie.load("token"),
+      },
+    })
+      .then((res) => res.json())
+      .catch((error) => {
+        history.push("/login");
+      });
+    if (!res.logged_in) {
```

Comment:
I notice this throws an error saying `res` is undefined when I try to visit the messenger page when not logged in. This is because when there's an error, `res` isn't assigned anything from the fetch. Since your errors aren't being returned from the server as JSON, calling `res.json()` throws an error. It would be better to only check `res.logged_in` if there's no error.

Reply:
Yes, actually after I finished this part, I tried the case you said. But I'm confused about the process and I'm going to rebuild it. Because when I am doing the socket part, I thought the logic in login/authentication is not very clear. I have already taken a note here. Thank you so much.